Support for optional SYNAPSE_DB_PORT env var

richvdh opened the pull request:
Add an env var to permit connecting to postgres on a different port

richvdh committed:
Support for optional SYNAPSE_DB_PORT env var

callahad commented on retention/cohort_analysis.py:
```diff
@@ -43,6 +43,7 @@ def get_conn(self):
         conn = connect(
             dbname=self.SYNAPSE_DB_DATABASE,
             user=self.SYNAPSE_DB_USERNAME,
+            port=os.environ.get("SYNAPSE_DB_PORT"),
```
Shouldn't this follow the same pattern as the other variables, which are set in the constructor immediately above?

richvdh replied:
well maybe, but that seems entirely redundant. I could refactor the others if you want?

callahad replied:
Either way, as long as it's consistent :)

richvdh replied:
fair enough.

I ended up refactoring it to build a DSN in the constructor (which is what `connect` does internally if you pass it a load of kwargs). WDYT?

richvdh committed:
Reduce duplication between Config.constructor and connect

Using a dsn here means we can do some validation early on, and then just have a
single thing to keep hold of.

callahad reviewed (approved):
Works for me
